### PR TITLE
Add support for readOnlyRootFilesystem (fixes #1402)

### LINKED ISSUE #1402: Expose the readOnlyRootFilesystem field that was added in cass-operator 1.22
In order to run Cassandra images with read only root FS (would only work for 4.1 and onwards), we need to expose the new `readOnlyRootFilesystem` which was added to cass-operator [here](https://github.com/k8ssandra/cass-operator/commit/cf277bc1c867376c57573340b1f545c69ad1ee30#diff-9a6ece13700a2568ec9915ade5b094433a489a30dcdb4aa75d75db4847d92b8aR274).

┆Issue is synchronized with this [Jira Story](https://datastax.jira.com/browse/K8OP-251) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by olim7t**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #1402

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by burmanm — APPROVED** [review]
Meh, no tests, but this should be quite straightforward here.

**Comment by adejanovski** [comment]
I've added some tests, but it seems like enabling this is creating issues with Stargate (don't ask me why...)

**Comment by adejanovski** [comment]
Unless it's the 4.1.5 upgrade that causes this.

**Comment by adejanovski** [comment]
Silly me, Stargate v1 doesn't work with 4.1...

File: test/e2e/suite_test.go [hunk 1]
```diff
@@ -946,7 +946,7 @@ func checkDatacenterReadOnlyRootFS(t *testing.T, ctx context.Context, key framew
 	dc := &cassdcapi.CassandraDatacenter{}
 	err := f.Get(ctx, key, dc)
 	require.NoError(t, err)
-	require.Equal(t, *dc.Spec.ReadOnlyRootFilesystem, *kc.Spec.Cassandra.ReadOnlyRootFilesystem, "expected datacenter %s to have readOnlyRootFilesystem like it is in the kc", key.Name)
+	require.Equal(t, ptr.Deref(dc.Spec.ReadOnlyRootFilesystem, false), ptr.Deref(kc.Spec.Cassandra.ReadOnlyRootFilesystem, false), "expected datacenter %s to have readOnlyRootFilesystem like it is in the kc", key.Name)
```
Comment (adejanovski) [inline test/e2e/suite_test.go:949]: Thanks @olim7t !

**Pull request merged**